lib/macho/macho_file.rb

```diff
@@ -556,7 +556,7 @@ def low_fileoff
 
       segments.each do |seg|
         sections(seg).each do |sect|
-          next if sect.size == 0
+          next if sect.size.zero?
```

Comment:
You're right, RuboCop misinterprets this because it cannot know what the contents of `sect` will be, but it assumes that `.size` is indicative of a container. I failed to realize this when commenting on this. I see several viable options:

- Put a `# rubocop:disable Style/ZeroLengthPredicate` comment at the end of the line to locally silence the misguided advice.
- Add an `empty?` convenience method to `Section[64]`. (It makes sense to me to speak of an empty section if the section has a size of zero bytes.)
- Slightly diverge from the C struct and rename the `size` attribute.